crates/ruff_dev/src/format_dev.rs

```diff
@@ -528,10 +545,19 @@ fn format_dir_entry(
         return Ok((Ok(Statistics::default()), file));
     }
 
-    let file = dir_entry.path().to_path_buf();
-    let options = options.to_py_format_options(&file);
+    let path = dir_entry.path().to_path_buf();
+    let mut options = options.to_py_format_options(&path);
+
+    let settings = resolver.resolve(&path, pyproject_config);
+    // That's a bad way of doing this but it's not worth doing something better for format_dev
+    if settings.line_length != LineLength::default() {
+        let line_width = LineWidth::try_from(settings.line_length.value())
+            .expect("Configured line length is too large for the formatter");
```

Comment:
i still don't like this divergence

Reply:
You can do `LineWidth::from(NonZeroU16::from(line_length)))`